### ISSUE TITLE
Login redirect complicates Firebase integration

### ISSUE THREAD

**Issue by dimitrovs** [auth0/lock#785]
This issue is related to https://github.com/auth0/lock/issues/514 , just a more complex scenario:

1) Single Page App with Firebase & Auth0 integration (only authenticated mode)
2) On load the app makes request to protected Firebase endpoint and catches authentication exception:
- More reliable than dealing with token expiry / local storage, if Firebase says we need to authenticate then we need to authenticate, no matter the underlying cause.
3) Firebase authentication exception triggers Auth0 Lock window display.
4) User provides credentials.
5) Auth0 performs authentication and redirects back to app.
6) Firebase request goes out before the Auth0 token has been exchanged with Firebase and this triggers the Auth0 Lock window to display again.

Basically the problem is that the Single Page App doesn't know when Auth0 is about to redirect the user or that a page load was the result of a redirect. The "authenticated" and "hash_parsed" events are not enough if there is another asynchronous event involved (Firebase auth attempt in this case). The only workaround currently is: ` { auth: { redirect: false }, autoclose: true }`

**Comment by hzalaz** [comment]
@dimitrovs you could "trigger" the protected request inside the `hash_parsed` when this callback supplies `null` meaning there was no auth redirect, if there is an auth redirect you save the token first and then perform the protected call

**Comment by dimitrovs** [comment]
Ok, this works, thanks.
